### DISCUSSION TITLE
Importing BaseModel from aws_lambda_powertools.utilities.parser vs pydantic directly?

### DISCUSSION THREAD

**Discussion by xkortex** [aws-powertools/powertools-lambda-python#4233] [General]
I noticed in the docs some examples use `from aws_lambda_powertools.utilities.parser import BaseModel`. Normally, I'd just use `from pydantic import BaseModel`. In my understanding, it is fairly atypical for a 3rd-party library to re-export another library's objects. I know this library relies on pydantic and with v2 there are several breaking changes, so I'm wondering if it is necessary to use `from aws_lambda_powertools.utilities.parser import BaseModel` in production code, or if it makes no difference to import directly from pydantic? The latter is far preferable (it confused me when I first dug into my new team's codebase). Is this some sort of shim?

We are currently on pydanticV1 but I want our team to rip off the bandaid sooner rather than later to upgrade to v2. 

This seems to be the [commit](https://github.com/aws-powertools/powertools-lambda-python/commit/b1a4821075d74debb5bdddccfb76769133256411) where there's a comment on the wildcard import/export, which is a bit of a headscratcher.

**Comment by heitorlessa** [comment]
no need at all - it was a poor judgement's decision to help non-developers use models among other things.

In a future v4/v5 version, I'd love to a custom Python's import hook to start addressing this in a backwards compatible fashion, deprecate it, and get rid of it.

If you've bandwidth, we'd love help in updating the docs to use a `pydantic` import instead.

**Reply by xkortex** [reply]
Great to know! And yeah I'd love to help update docs but I need to check with my company.  

> `get rid of it.` 

it meaning pydantic1?

Python has some of the most powerful import machinery of any language I've ever encountered, I'm sure there exists a way to polyfill ALP such that the pydantic version is (mostly) irrelevant.

**Reply by heitorlessa** [reply]
yeah, we've started working on the next major version (v3) - [RFC](https://github.com/aws-powertools/powertools-lambda-python/issues/4189) - and we will remove pydantic v1 support _(today we support both)_.

Couldn't agree more on import machinery. It is possible, just haven't had a chance to craft one to remap imports, as breaking change is a huge deal for us.
